types/minio/index.d.ts

```diff
@@ -187,7 +191,12 @@ export class Client {
 
     // Other
     newPostPolicy(): PostPolicy;
-    setRequestOptions(otpions: AgentOptions): void;
+    setRequestOptions(options: AgentOptions): void;
+
+    // Minio extensions that aren't necessary present for Amazon S3 compatible storage servers
+    extensions: {
+        listObjectsV2WithMetadata(bucketName: string, prefix?: string, recursive?: boolean, startAfter?: string): BucketStream<BucketItemWithMetadata>;
+    };
```

Comment:
Are you familiar with what cases `extensions` are available? I've tried reading the documentation but couldn't find any notes on their "extension" system.

Reply:
I found something. It's [in the doc](https://docs.min.io/docs/javascript-client-api-reference#listObjectsV2WithMetadata) below `listObjectsV2WithMetadata` method:

```
var stream = minioClient.extensions.listObjectsV2WithMetadata('mybucket','', true,'')
stream.on('data', function(obj) { console.log(obj) } )
stream.on('error', function(err) { console.log(err) } )
```

Reply:
It seems it's some part of server configuration - https://docs.min.io/docs/minio-bucket-replication-guide.html
In heading MinIO Extension
🤔

Reply:
> Are you familiar with what cases `extensions` are available? I've tried reading the documentation but couldn't find any notes on their "extension" system.

From the client's perspective, this property (the *extensions*) is available as a lazy getter (as seen [here](https://github.com/minio/minio-js/blob/master/src/main/minio.js#L3047)). Actual extensions are defined in a separate file - [here](https://github.com/minio/minio-js/blob/master/src/main/extensions.js). No server checks are performed on the backend side to see if they are indeed supported. Under the hood, this method is using the S3 *listObjectsV2* API with an additional query parameter `metadata=true`.

As for the server configuration goes - I'm not aware if there are any specific ones, that must be set before using this functionality. Using a pre-built Docker image, this works just fine out of the box.